tests/test_openclhpp.cpp

```diff
@@ -4517,4 +4517,37 @@ void testgetObjectInfo() {
     TEST_ASSERT_EQUAL(type, CL_GL_OBJECT_BUFFER);
     TEST_ASSERT_EQUAL(bufobj, 0);
 }
+#if CL_HPP_TARGET_OPENCL_VERSION >= 110
+static cl_mem clCreateSubBuffer_testCreateSubBuffer(
+    cl_mem buffer, cl_mem_flags flags, cl_buffer_create_type buffer_create_type,
+    const void *buffer_create_info, cl_int *errcode_ret, int num_calls) {
+    (void)errcode_ret;
+    TEST_ASSERT_EQUAL(make_mem(0), buffer);
+    TEST_ASSERT_EQUAL(0, flags);
+    TEST_ASSERT_EQUAL(0, buffer_create_type);
+    TEST_ASSERT_EQUAL_PTR(nullptr, buffer_create_info);
+    TEST_ASSERT_EQUAL(0, num_calls);
+
+    return make_mem(1);
+}
+
+void testCreateSubBuffer() {
+#ifndef CL_HPP_ENABLE_EXCEPTIONS
+    cl_mem_flags flags = 0;
+    cl_buffer_create_type buffer_create_type = 0;
+    const void *buffer_create_info = nullptr;
+    cl_int *err = nullptr;
+    static cl::Buffer ret;
+
+    clCreateSubBuffer_StubWithCallback(clCreateSubBuffer_testCreateSubBuffer);
+    ret = bufferPool[0].createSubBuffer(flags, buffer_create_type,
+                                        buffer_create_info, err);
+
+    TEST_ASSERT_EQUAL_PTR(make_mem(1), ret());
+    ret() = nullptr;
```

Comment:
Please make this variable non-static:

```suggestion
    clCreateSubBuffer_StubWithCallback(clCreateSubBuffer_testCreateSubBuffer);
    clReleaseMemObject_ExpectAndReturn(make_mem(1), CL_SUCCESS);
    cl::Buffer ret = bufferPool[0].createSubBuffer(flags, buffer_create_type,
                                        buffer_create_info, err);

    TEST_ASSERT_EQUAL_PTR(make_mem(1), ret());
```